api/proto/config/appconfig.proto

```diff
@@ -100,6 +107,9 @@ message AppInstanceConfig {
   // Note that since the name volumeRef was used before and deprecated
   // python protobuf seems to require that we use a different name.
   repeated VolumeRef volumeRefList = 16;
+
+  // metadata type to use for app
```

Comment:
Question: do we envision that we might provide meta-data even if there is no user-data supplied for the app instance? If so we'd need a MetaDataNone. Currently we only provide the cdrom (with meta-data and user-data) if there is some user-data.

Reply:
I do not want to break existing flow with default 0 inside. So, I added comment `if provided inside userData`.